timon/probes.py

```diff
@@ -79,6 +79,7 @@ async def run(self):
             await self.probe_action()
             logger.debug("finished probe %r", name)
         except Exception:
+            self.status = "ERROR"
```

Comment:
I am not sure this is useful.
The question is whether the whole 'task' performing the probe will terminate because of the raise before any status is returned or not.
In the latter case `self.status` would be changed, but never be used.

Without proving (small test scenario) that it is useful I'd prefer to not add this line, whichmight suggest something, that might not be happening.

Reply:
I can remove it, but, i think it's not a bad thing. The default status is unknown, in case of a strange case happen a day. This day, we will not be notified because of the "UNKNOWN" status (that will not be correct). I prefer set error, because it's an error.

If you prefer delete this line and pray this never happen, i can delete it. But i don't agree with the fact that because a thing might not be happening, this will never happen.

Reply:
i think you did not at all understand my concern. I am not 100% convinced that this line will do what you think.

make a test (add an error in the try statement (e.g. 1/0) to show, that 
```
self.status = "ERROR"
raise
```

will really result in a probe result with error, that will be saved in the json file. 
I don't know the code well enough to be sure.

Often raise might stop everything, so that it doesn't matter whether you set state to 'ERROR" or to "VALUE_THAT_NOBODY_WILL_EVER_SEE" because the value will no more be used.

My comment meant.

Please make sure, that what  you do will really result in a probe state == "ERROR" that will be saved and not jsut a probe state = "ERROR" that will not be osed by the consecutive code.

Reply:
Yeah i understood, i remove that line

Reply:
not necessarily. if you can make a test and see that the state is saved it is a good idea, but we have to test